File: backend/app/models/supplier.py
from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database.base import Base


class Supplier(Base):
    __tablename__ = "suppliers"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True,
    )

    supplier_code: Mapped[str] = mapped_column(
        String(20),
        unique=True,
        nullable=False,
        index=True,
    )

    company_name: Mapped[str] = mapped_column(
        String(200),
        nullable=False,
    )

    contact_person: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    email: Mapped[str] = mapped_column(
        String(150),
        unique=True,
        nullable=False,
    )

    phone: Mapped[str] = mapped_column(
        String(20),
        nullable=False,
    )

    gst_number: Mapped[str] = mapped_column(
        String(30),
        unique=True,
        nullable=False,
    )

    address: Mapped[str] = mapped_column(
        String(500),
        nullable=False,
    )

    city: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    state: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    pincode: Mapped[str] = mapped_column(
        String(10),
        nullable=False,
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        default=True,
        nullable=False,
    )

    created_by: Mapped[int] = mapped_column(
        ForeignKey("users.id"),
        nullable=False,
    )

    created_at: Mapped[DateTime] = mapped_column(
        DateTime,
        server_default=func.now(),
    )

    updated_at: Mapped[DateTime] = mapped_column(
        DateTime,
        server_default=func.now(),
        onupdate=func.now(),
    )

File: frontend/src/types/supplier.ts
export interface Supplier {
    id: number;
  
    supplier_code: string;
    company_name: string;
    contact_person: string;
  
    email: string;
    phone: string;
  
    gst_number: string;
  
    address: string;
    city: string;
    state: string;
    pincode: string;
  
    is_active: boolean;
  
    created_at: string;
    updated_at: string;
  }
  
  
  export interface SupplierCreatePayload {
    supplier_code: string;
    company_name: string;
    contact_person: string;
  
    email: string;
    phone: string;
  
    gst_number: string;
  
    address: string;
    city: string;
    state: string;
    pincode: string;
  
    is_active: boolean;
  }
  
  
  export type SupplierUpdatePayload =
    SupplierCreatePayload;